kentikapi/models/user.go

```diff
@@ -2,24 +2,45 @@ package models
 
 import "time"
 
-type GetAllUsersResponse struct {
-	Users []User `json:"users"`
+// User model.
+type User struct {
+	// read-write properties (can be updated in update call)
+	Username     string
+	UserFullName string
+	UserEmail    string
+	// Role of the user, e.g. "Member", "Administrator", "Super Administrator".
+	Role         string
+	EmailService bool
+	EmailProduct bool
+
+	// read-only properties (can't be updated in update call)
+	ID           ID
+	LastLogin    *time.Time
+	CreatedDate  time.Time
+	UpdatedDate  time.Time
+	CompanyID    ID
+	UserAPIToken *string
 }
 
-type GetUserResponse struct {
-	User User `json:"user"`
+// UserRequiredFields is subset of User fields required to create a User.
+type UserRequiredFields struct {
+	Username     string
+	UserFullName string
+	UserEmail    string
+	// Role of the user, e.g. "Member", "Administrator", "Super Administrator".
+	Role         string
+	EmailService bool
+	EmailProduct bool
 }
 
-type User struct {
-	ID           ID         `json:"id,string"`
-	Username     string     `json:"username"`
-	UserFullName string     `json:"user_full_name"`
-	UserEmail    string     `json:"user_email"`
-	Role         string     `json:"role"`
-	EmailService bool       `json:"email_service"`
-	EmailProduct bool       `json:"email_product"`
-	LastLogin    *time.Time `json:"last_login"`
-	CreatedDate  time.Time  `json:"created_date"`
-	UpdatedDate  time.Time  `json:"updated_date"`
-	CompanyID    ID         `json:"company_id,string"`
+// NewUser creates a new User with all required fields set.
+func NewUser(u UserRequiredFields) *User {
```

Comment:
https://kb.kentik.com/v0/Ec06.htm#Ec06-User_Create says that all those fields are required.
In fact, only UserEmail is required by the API, rest are optional.
In Python library we treat password and role as optional. 

I implemented it to behave just like API documentation. @mateuszmidor @lukaa12 What do you think?

BTW I removed `user_password` field, because it seems that there is no such thing in the API.

Reply:
Stayed with all required choice.